### PR TITLE
Add pod index label docs for KEP-4017

### PR DESCRIPTION
**Pull request by danielvegamyhre**
The purpose of these changes are to add documentation for the new pod index label being added for StatefulSets and Indexed Jobs (see [KEP-4017](https://github.com/kubernetes/enhancements/pull/4019)).

cc @ahg-g @alculquicondor @soltysh

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 1]
```diff
@@ -238,6 +239,13 @@ it adds a label, `statefulset.kubernetes.io/pod-name`, that is set to the name o
 the Pod. This label allows you to attach a Service to a specific Pod in
 the StatefulSet.
 
+### Pod Index Label
```
[the remark this answers is not in the document]
Comment (danielvegamyhre) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:242]: done

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 2]
```diff
@@ -283,6 +283,7 @@ Jobs with _fixed completion count_ - that is, jobs that have non null
 - `Indexed`: the Pods of a Job get an associated completion index from 0 to
   `.spec.completions-1`. The index is available through three mechanisms:
   - The Pod annotation `batch.kubernetes.io/job-completion-index`.
+  - The Pod label: `batch.kubernetes.io/job-completion-index`.
```
[the remark this answers is not in the document]
Comment (danielvegamyhre) [reply content/en/docs/concepts/workloads/controllers/job.md]: added

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 3]
```diff
@@ -283,6 +283,7 @@ Jobs with _fixed completion count_ - that is, jobs that have non null
 - `Indexed`: the Pods of a Job get an associated completion index from 0 to
   `.spec.completions-1`. The index is available through three mechanisms:
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/job.md:284]:
```suggestion
  `.spec.completions-1`. The index is available through four mechanisms:
```
?

File: content/en/docs/concepts/workloads/controllers/job.md [hunk 4]
```diff
@@ -283,6 +283,7 @@ Jobs with _fixed completion count_ - that is, jobs that have non null
...
   - The Pod annotation `batch.kubernetes.io/job-completion-index`.
+  - The pod label `batch.kubernetes.io/job-completion-index` (for v1.28 and later).
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/job.md:286]: Please capitalize “pod” consistently over these 2 lines and ideally over the entire list.

File: content/en/docs/concepts/workloads/controllers/statefulset.md [hunk 5]
```diff
@@ -238,6 +239,15 @@ it adds a label, `statefulset.kubernetes.io/pod-name`, that is set to the name o
...
+
+{{< feature-state for_k8s_version="v1.28" state="beta" >}}
+
+When the StatefulSet {{<glossary_tooltip text="controller" term_id="controller">}} creates a Pod,
+it adds a label, `apps.kubernetes.io/pod-index`, that is set to the ordinal index of
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/statefulset.md:247]: Avoid a risk that people infer “it” to be ”the new Pod” here.
```suggestion
the new Pod is labelled with `apps.kubernetes.io/pod-index`. The value of this label is the ordinal index of
```

**Review by sftim — COMMENTED** [review]
Thanks

Some feedback

Please also make sure that https://kubernetes.io/docs/reference/labels-annotations-taints/ is up to date.

File: content/en/docs/concepts/workloads/controllers/job.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielvegamyhre) [reply content/en/docs/concepts/workloads/controllers/job.md]: Updated https://kubernetes.io/docs/reference/labels-annotations-taints/  and added a note about the feature gate in the relevant sections of both job docs and statefulset docs.

File: content/en/docs/concepts/workloads/controllers/statefulset.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielvegamyhre) [reply content/en/docs/concepts/workloads/controllers/statefulset.md]: done

File: content/en/docs/concepts/workloads/controllers/job.md (hunk 3 above)
Comment (danielvegamyhre) [reply content/en/docs/concepts/workloads/controllers/job.md]: done

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 6]
```diff
@@ -1035,13 +1035,13 @@ by the cloud-controller-manager.
 
 ### batch.kubernetes.io/job-completion-index
 
-Type: Annotation
+Type: Label and Annotation
```
Comment (alculquicondor) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1038]: Also add an entry for the new statefulset label

File: content/en/docs/concepts/workloads/controllers/statefulset.md (hunk 5 above)
Comment (sftim) [reply content/en/docs/concepts/workloads/controllers/statefulset.md:247]: I unresolved this conversation as my concern stands.

Comment (danielvegamyhre) [reply content/en/docs/concepts/workloads/controllers/statefulset.md]: sorry about that I thought I hit "commit suggestion" already but I guess not. Thanks!

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 6 above)
Comment (danielvegamyhre) [reply content/en/docs/reference/labels-annotations-taints/_index.md:1052]: ah of course - added!

**Comment by alculquicondor** [comment]
LGTM as job approver

**Review by sftim — COMMENTED** [review]
One thing to fix, and a nit that is entirely optional to redo.

**Comment by sftim** [comment]
Changes since https://github.com/kubernetes/website/pull/42059#issuecomment-1644066308 are trivial
LGTM

/hold

Doesn't this need to target the _dev-1.28_ branch though (I hadn't spotted that before)?

**Comment by danielvegamyhre** [comment]
Resolved merge conflicts with `dev-1.28` branch.

**Comment by danielvegamyhre** [comment]
Oh jeez, after I changed the target to dev-1.28 and resolved the merge conflict somehow hundreds of unrelated commits are included on this PR...

**Comment by danielvegamyhre** [comment]
Copied over the approved changes into a new PR targeting dev-1.28: https://github.com/kubernetes/website/pull/42139 @sftim  @alculquicondor 

Closing this PR now.

**Pull request closed without merging**